src/site/_collections/shows.js

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const CacheAsset = require('@11ty/eleventy-cache-assets');
+const {defaultLocale} = require('../../../shared/locale');
+
+/** @type ShowsData */
+const showsData = require('../_data/showsData.json');
+
+/**
+ * Returns all authors with their posts.
+ *
+ * @return {Promise<Shows>}
+ */
+module.exports = async () => {
+  /** @type Shows */
+  const shows = {};
+  const keys = Object.keys(showsData);
+
+  for (const key of keys) {
+    const showData = showsData[key];
+    const href = `/shows/${key}/`;
+    const url = `https://storage.googleapis.com/web-dev-uploads/youtube/${showData.playlistId}.json`;
+    const elements = (
+      await CacheAsset(url, {
+        duration: '6h',
+        type: 'json',
+      }).catch((e) => {
+        console.warn(e);
+        return [];
+      })
+    ).map((v) => {
+      v.date = new Date(v.date);
+      v.data.date = new Date(v.data.date);
+      v.url = `${href}${v.data.videoId}/`;
+      v.data.lang = defaultLocale;
+      return v;
+    });
+    if (elements.length === 0) {
+      continue;
+    }
+
+    /** @type ShowsItem */
+    const show = {
+      ...showData,
+      data: {
+        date: elements[elements.length - 1].date,
+        hero: elements[0].data.thumbnail,
+      },
+      description: `i18n.shows.${key}.description`,
+      elements,
+      href,
+      key,
+      title: `i18n.shows.${key}.title`,
+      url: href,
+    };
+
+    // Limit posts for percy
+    if (process.env.PERCY) {
+      show.elements = show.elements.slice(-6);
+    }
+
+    // Set created on date and updated date
+    if (show.elements.length > 0) {
+      show.data.date = show.elements.slice(-1).pop().data.date;
+      const updated = show.elements.slice(0, 1).pop().data.date;
+      if (show.data.date !== updated) {
+        show.data.updated = updated;
+      }
+    }
```

Comment:
I'm not sure, but this could/would be used by the firebase indexing and lets us say when a series has been updated (when a new video has been added to the playlist)

Reply:
Can you add a comment explaining that? I recall asking the same question about a similar block duplicated elsewhere in the code

Reply:
Done!